runtime/bcutil/cfreader.c

```diff
@@ -2480,6 +2482,12 @@ j9bcutil_readClassFileBytes(J9PortLibrary *portLib,
 		return -1;
 	}
 
+	/* TODO right its checking if major version is newer than JAVA15. Newer classfile
+	 * versions will not require to inject the identityObject */
+	if (classfile->majorVersion > VALUE_TYPES_MAJOR_VERSION) {
```

Comment:
My assumption here is that only old classfiles need the interface injection. The new class files will be compiled with the interface.

Reply:
By javac, probably.  There are a lot of other class generators out there that may be incorrect though